force-app/main/default/classes/RD2_ChangeHistoryController.cls

```diff
@@ -0,0 +1,12 @@
+public with sharing class RD2_ChangeHistoryController {
+
+    @AuraEnabled
+    public static RD2_ChangeHistoryView getChangeHistory(Id recurringDonationId, Integer recordLimit, String filter) {
+        RD2_ChangeHistoryService service = new RD2_ChangeHistoryService();
```

Comment:
Since this service is a collaborator for this class, could we use the lazy load singleton pattern we have used elsewhere to make it easier to mock the service directly in test classes in the future? Example below:

`@TestVisible
    private static RD2_ChangeHistoryService changeHistoryService {
        get {
            if (changeHistoryService == null) {
                changeHistoryService = new RD2_ChangeHistoryService();
            }
            return changeHistoryService;
        }
        set;
    }`

Reply:
I think that's a little bit more than we'd need there, every method in this file should be `@AuraEnabled public static`, and no two methods in this file should get called in the same transaction.